drivers/espi/espi_it8xxx2.c

```diff
@@ -687,22 +696,130 @@ static int espi_it8xxx2_write_lpc_request(const struct device *dev,
 }
 
 #ifdef CONFIG_ESPI_OOB_CHANNEL
+/* eSPI cycle type field */
+#define ESPI_OOB_CYCLE_TYPE          0x21
+#define ESPI_OOB_TAG                 0x00
+#define ESPI_OOB_TIMEOUT_MS          200
+
+struct espi_oob_msg_packet {
+	uint8_t cycle_type;
+	uint8_t tag_len_bit_11_8;
+	uint8_t len_bit_7_0;
+	uint8_t data_byte[ESPI_IT8XXX2_OOB_MAX_PAYLOAD_SIZE];
+};
+
 static int espi_it8xxx2_send_oob(const struct device *dev,
 				struct espi_oob_packet *pckt)
 {
-	/* TODO: implement me... */
-	ARG_UNUSED(dev);
+	const struct espi_it8xxx2_config *const config = DRV_CONFIG(dev);
+	struct espi_slave_regs *const slave_reg =
+		(struct espi_slave_regs *)config->base_espi_slave;
+	struct espi_queue1_regs *const queue1_reg =
+		(struct espi_queue1_regs *)config->base_espi_queue1;
+	struct espi_oob_msg_packet *oob_pckt =
+		(struct espi_oob_msg_packet *)pckt->buf;
+	uint16_t oob_msg_len = oob_pckt->len_bit_7_0 |
+				((oob_pckt->tag_len_bit_11_8 & 0xf) << 8);
```

Comment:
You should also verify that `pckt->len == oob_msg_len + 3`.

Reply:
Done.